pkg/api/types.go

```diff
@@ -2119,34 +2119,121 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
-// Type and constants for component health validation.
+// Type and constants for component type.
+type ComponentType string
+
+// ComponentType constants define some known types of components.
+// Each type may have multiple instances, as long as their names are unique.
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
+	ComponentScheduler         ComponentType = "scheduler" // TODO: move mesos-specific component type & validation to contrib
+)
+
+// ComponentSpec describes the attributes that a component is created with.
+type ComponentSpec struct {
+	// Type of the component
+	Type ComponentType `json:"type"`
+	// Address to reach the component (scheme, host, port, path). Address paths determined by component type.
+	Address string `json:"address,omitempty"`
+}
+
+type ComponentPhase string
+
+// These are the valid phases of a component.
+const (
+	// ComponentPending means the component has been created, but is still starting up.
```

Comment:
Does pending matter, or would clients of this API just care about readiness (what you call healthy)?